aws/request/http_request.go

```diff
@@ -24,3 +24,13 @@ func copyHTTPRequest(r *http.Request, body io.ReadCloser) *http.Request {
 		Cancel: r.Cancel,
 	}
 }
+
+// IsCanceled checks the cancel channel to see if it has been closed
+func (r *Request) IsCanceled() bool {
```

Comment:
I don't think these methods are necessary. Also I don't think they capture if the `Transport.CancelRequest` was the reason the request being canceled. They provide a false sense of information the SDK may not have.